### ISSUE TITLE
Monthly Chat Agenda November 2022 (2022-11-07 and 2022-11-21)

### ISSUE THREAD

**Issue by dune73** [coreruleset/coreruleset#2824]
This is the Agenda for the Monthly CRS Chat.

The general chat is going to happen on https://owasp.slack.com in the channel #coreruleset on Monday, 2022-11-07, at 20:30 CET. That's the 1st Monday of the month. A separate issue chat is happening at the same location, same time on Monday, 2022-11-21. That's the 3rd Monday of the month. Please note that we have a [CRS calendar](https://calendar.google.com/calendar/u/0/embed?src=c_glsll5etlmcb3it0k433fcl5rc@group.calendar.google.com) (maintained by @fzipi).

Archived previous meetings and their decision are [here](https://github.com/coreruleset/project-chat-archive/blob/master/README.md).

The monthly meeting follows a new agenda structure and focuses more on project overview and less on individual PRs. See https://github.com/coreruleset/coreruleset/wiki/DevRetreat22WorkshopRoadmap

### Outside development

* [Blog about GeoIP & ASN](https://www.netnea.com/cms/2022/10/12/using-geoip-information-together-with-modsecurity/) by Christian Folini / @dune73
* [Portrait of CRS developer Andrea Menin / @theMiddle](https://coreruleset.org/20221018/meet-the-crs-team-andrea-the-musical-man-in-the-middle/)
* Successful CRS Developer Retreat in Italy. Blog Post pending, notes at https://github.com/coreruleset/coreruleset/wiki/Dev-Retreat-2022
* [Next ModSec / CRS webcast](https://www.meetup.com/meetup-group-ungjkskv/events/289074360/) by Christian Folini: Nov 14

#### Rules
* Bug progress with over 30 bug bounty fixed.
* We had [a report](https://github.com/coreruleset/coreruleset/issues/2976) on an issue that lead to all regular expressions failing to compile on Ubuntu 22.04, which, effectively, disabled the WAF. The culprit appears to have been a bad package, possibly nginx-mod-lua, from a specific apt repository.

#### CRS Sandbox

* Not much movement
* Talks about retiring the elastic monitoring for a lack of use (dune73 thinks it would take a custom sandbox dashboard to make it interesting)

#### CRS Bug Bounty and Security
* Decision to track security findings, namely those received via email, as issues in a separate private repository
* bug bounty findings all covered, but checks reveal not all tests are successful. [Link to private checklist](https://github.com/coreruleset/bug-bounty-tests)
* see below for discussion.

#### Plugins
*  [Extensive discussion](https://github.com/coreruleset/nextcloud-rule-exclusions-plugin/issues/2) on allowed method variable in nextcloud plugin.
* WordPress Plugin: updated the exclusion rules and planning to in Labels
bookmark Meeting Agenda
cluding top used WP-Plugin. Little other movement.

#### Documentation and Public Relations
* Release of first dev portrait (see above), 2nd in the making
* Blog post about dev retreat in the making.

#### Project Administration and Sponsor relationships
* OWASP HQ accepted monthly payment for administration
* Several upcoming sponsorship announcements in the making

#### Tools
* go-ftw fixed several bugs during retreat
* crs-toolchain to replace perl based regex-assemble and more tooling (regex-assemble part now on go implementation of regex-assemble which allows more powerful configuration). 
* **Pending question**: Migration of regex-assemble to crs-toolchain

#### Testing incl. Seaweed and many future plans
* Currently on hold

#### Containers
* Decision during retreat to let go of the naked ModSec container and to concentrate on a CRS container.

#### CRS Status Page
* We decided to aim for a MVP where we check 80% of our PL1 rules.
* We are currently on 70% of PL1 rules for CRS v4.
*  More information [here](https://github.com/coreruleset/coreruleset/wiki/DevRetreat22ProjectStatusPage)

## Project Decisions

* How to cover for bug bounty findings that have negative tests (-> false negatives!)?

* When and how to migrate to CRS toolchain?

* Release plan for CRS 4?

### PRs that have been merged since the last meeting

* #2970
* #2969
* #2968
* #2967
* #2965
* #2964
* #2952
* #2962
* #2877
* #2894
* #2938
* #2932
* #2963
* #2956
* #2960
* #2958
* #2937
* #2832
* #2921
* #2955
* #2951
* #2907
* #2904
* #2935
* #2923
* #2925
* #2928
* #2931
* #2927
* #2922

We merged 30 PRs since the last monthly project chat.

### Open PRs marked *DRAFT* or *work in progress* or *needs action*

* #2974
* #2975
* #2966
* #2954
* #2972
* #2971
* #2948
* #2878
* #2945
* #2957
* #2949
* #2950
* #2941
* #2947
* #2946
* #2942
* #2939
* #2933
* #2637
* #2573
* #2791
* #2666

### Tasks until CRS v4

- [x] Merge as many PRs as we can (before toolchain migration)
- [x] Toolchain migration
- [ ] **MUST** Close _all_ BB findings, recheck (stats right now: 73.33% blocked at PL1, 88% at PL2, 88% at PL3, 95% at PL4)
- [ ] **MUST** close Shivam Bathla's bypasses
- [ ] **MUST** list updates: still critical with [20 open issues, 13 closed](https://github.com/coreruleset/coreruleset/issues?q=is%3Aissue+is%3Aopen+label%3A%22%3Azap%3A+list+update%22)
- [ ] **MUST** Document <Location> and phase 1 rules
- [ ] **MUST** Split tasks in ready for RC2 vs GA
- [ ] **MUST** CI: integrate all checks that @airween has developed
- [ ] **CAN** WordPress plugin updates
- [ ] **CAN** Setup CI for plugins (nice2have)
- [ ] **CAN** Add TX:/crs_arg_.*/ to all rules that check ARGS  (-> check performance impact first)

### Status of issues covered last month

* Issue slot 1: #FIXME
* Issue slot 2: #FIXME
* Issue slot 3: #FIXME
* Issue slot 4: #FIXME
* Issue slot 5: #FIXME
* Issue slot 6: #FIXME
* Issue slot 7: #FIXME
* Issue slot 8: #FIXME
* Issue slot 9: #FIXME
* Issue slot 10: #FIXME

**Stats**

* Covered in chat: FIXME
* Closed: FIXME
* Pending: FIXME

### This month's issues

There are FIXME open issues at the beginning of the issue chat.

We generally cover 10 issues per month in a separate issue meeting. Add them as you see fit.

* Issue slot 1: #FIXME
* Issue slot 2: #FIXME
* Issue slot 3: #FIXME
* Issue slot 4: #FIXME
* Issue slot 5: #FIXME
* Issue slot 6: #FIXME
* Issue slot 7: #FIXME
* Issue slot 8: #FIXME
* Issue slot 9: #FIXME
* Issue slot 10: #FIXME

## How to get to our slack and join the meeting?

If you are not yet on the OWASP Slack, here is your invite: https://owasp.org/slack/invite .

Everybody is welcome to join our community chat.

**Comment by dune73** [comment]
### Decisions
* Bug Bounty Findings, 2nd pass: Every dev is asked to look at 5 bug bounty findings that are still on 200 despite PRs and investigate (-> Issue -> PR). See https://github.com/coreruleset/bug-bounty-tests for list. (Test ID means number of curl call of finding.)
* @theseion finishes the toolchain. Once he is done, @lifeforms will give the signal for migration, also for migration of CI. The old regex-assemble will be replaced with a script that fails with notice to run toolchain instead.
* We are in no position to do a CRS4 release plan. Too many things open on bug bounty findings, 30 PR and toolchain. Discussion postponed to meeting in 2 weeks.

**Comment by franbuehler** [comment]
## Decisions Nov 21

* deny at phase:1 is a problem for NGINX -> we need to explain to people they should stick to phase 2 blocking and they should not activate early blocking ->  documentation: extend early blocking doc with phase 1 and its caveats/problems, maybe blog post
* boosting performance removing unneeded variables from rules and maybe remove unnecessary PL-skip rules in every rule file -> will be part of a larger performance campaign for 4.1 -> in the meantime: implement as plugin
* @franbuehler will work on Shivam Bathla's new bypasses, should be finished in 2 weeks
* @fzipi will work on list updates, but needs some help. It's a big task. He will tag people on the list update.
* @theseion will focus on the BB issues, but everyone is invited to help. @theMiddleBlue will help on "curl failed issues"
